src/pages/Dashboard.tsx

```diff
@@ -81,10 +102,37 @@ const Dashboard = () => {
         }
     }
 
+    const createNewGroup = async (groupName: string) => {
+        try {
+            await addUserToGroup(TEST_USER, groupName);
+            setUserGroups(currentUserGroups.concat(groupName));
```

Comment:
Don't mutate the state directly, create a copy first or do (prevUserGroups) =>  [...prevUserGroups, groupName]